libgnucash/app-utils/business-options.scm

```diff
@@ -73,6 +75,8 @@
 (define gnc:*company-url* (N_ "Company Website URL"))
 (define gnc:*company-email* (N_ "Company Email Address"))
 (define gnc:*company-contact* (N_ "Company Contact Person"))
+(define gnc:*company-iban* (N_ "Company IBAN"))
+(define gnc:*company-bic* (N_ "Company BIC"))
```

Comment:
If no IBAN is provided, you need BIC + traditional Bank Account Number for a bank transaction, so I would add it here, too. Optionally also the name (including branch) of the bank.
It is common to add this infos to the footer of a business letter.

[IBAN](https://en.wikipedia.org/wiki/International_Bank_Account_Number#Basic_Bank_Account_Number)
SWIFT-BIC: [ISO 9362](https://en.wikipedia.org/wiki/ISO_9362) and links.

Reply:
You mean we offer the new options IBAN, BIC, accnumber, and:
1. QR encoder picks up IBAN + BIC
2. invoice report prints IBAN & BIC+accnumber?

Otherwise what's the purpose of having acc-number?

Reply:
I made a sketch with many of the relations in [Payment Routing](https://wiki.gnucash.org/wiki/Payment_Routing).

Reply:
In the example in https://wiki.gnucash.org/wiki/File:Invoice_Option_Display.png I offer 3 bank connections. So you would better create a QR-code for each of them and disply it  together with the textual form at the bottom of the page:
```
Pay via
Bank: DeuBa VoBa SpK
IBAN: …
QR: …
```

Reply:
I seriously don't think it's reasonable to require or display multiple QR codes.

Reply:
> I seriously don't think it's reasonable to require or display multiple QR codes.

Scheme is list processing. Take the invoice amount and the list of IBANS (in the future) from the book properties to generate the payment section of the invoice.

The other problem: If there is a payment term like 2-10-30 (2% discount on payments in ten days), you have to generate 2 lists: for 98% and 100% of the invoice amount.

Reply:
> In the example in https://wiki.gnucash.org/wiki/File:Invoice_Option_Display.png I offer 3 bank connections. So you would better create a QR-code for each of them and display it together with the textual form at the bottom of the page:
> 
> ```
> Pay via
> Bank: DeuBa VoBa SpK
> IBAN: …
> QR: …
> ```

A few thoughts to avoid dragging this proposal forever.

I would propose to to focus on implementing one specific scenario first and consider expanding it in future iterations based on user feedback. From reading the comments on this PR it looks like the current scenario is to implement EPC QR. I think that's a good start and it requires an IBAN account option and can have BIC as optional parameter. I think that's a good goal to reach first.

Before we go and implement variants with multiple bank account identifiers I would ask around for real world examples of invoices printing a QR code. IMO even if a company has multiple bank accounts, you can still pick one as the preferred one and use that for a QR code. Adding multiple QR codes to an invoice sounds fairly clumsy to me and a source of confusion to the the customer. I'm aware I say that while I have never ever seen a real invoice with QR codes yet so I'd like to have real world examples (even if obfuscated).

I also don't know if the comments about national bank account numbers and bank identifiers apply to this PR. The EPC QR specifically states it uses IBAN numbers,. So while it may be nice to print those traditional numbers and identifiers on your invoice they shouldn't be used to print QR codes. Unless there's another standard supporting that use case. Again if so, that could be a future improvement. Adding the ability to print the traditional bank numbers on your invoices should also be done in a separate PR as it will probably take a few layout iterations to work for many cases.

Reply:
> The other problem: If there is a payment term like 2-10-30 (2% discount on payments in ten days), you have to generate 2 lists: for 98% and 100% of the invoice amount.

Same thing here. I don't know how this is handled in the real world today. I would probably not put QR codes at all if the payment amount depends on other factors or I would add one single QR code for the most common case. Real world samples would help a lot in this area.

Reply:
I would also now only implement something like
```
<bankconnections>
  <bankdetails>
    <bankname></bankname>
    <IBAN></IBAN>
    <BIC></BIC>
  </bankdetails>
</bankconnections>

```
with one instance in book company info.

About the layout.
1. AFAIK the size of a stamp is sufficient . In CH 46mm, 5mm border
2. I would put it below the sum as it is part of the content.
3. Alternative it can be printed as part of a regular transfer form 
[zahlschein.pdf](https://github.com/Gnucash/gnucash/files/5827916/zahlschein.pdf)
at the bottom or on the next page.

Reply:
@fellen you know full well that the set of company options is currently a global pref and is out of scope for this PR. Please liaise with @jralls or @gjanssens how to move company options from global to book. This is not my game.

Reply:
> the set of company options is currently a global pref

No,
![image](https://user-images.githubusercontent.com/6572538/104875380-75e58380-5955-11eb-9632-67457072c7b4.png)

Reply:
This is still a major change in the datafile format, therefore out of scope for this bug.

Reply:
1. it is no bug, but an enhancement request.
2. There is a new dependency.
3. There is a change in the file structure.
So IMHO you should add the ability to read a file with IBAN … to maint and the rest in master.

Reply:
@fellen That's absurd. The business options exist solely to create invoices. This use of IBAN/BIC is solely to put a QR on invoices. Business options are stored in KVP so no format changes and no compatibility worries.

The only issue with this PR is about the dependency.